lib/config.js

```diff
@@ -115,3 +115,136 @@ Config.prototype.checkForUnknownRuleFieldNames = function(errors) {
     }
   });
 };
+
+Config.prototype.checkForMisconfiguredRules = function(errors) {
+  checkRulesAreSorted(this.rules, errors);
+  checkRulesByReaction(this.rules, errors);
+};
+
+function checkRulesAreSorted(rules, errors) {
+  var sorted = sortIfUnsorted(rules, compareRules),
+      NUM_SPACES = 2;
+
+  if (sorted) {
+    errors.push('rules are not sorted; expected: ' +
+      JSON.stringify(sorted, null, NUM_SPACES).replace(/\n/g, '\n  '));
+  }
+}
+
+function checkRulesByReaction(rules, errors) {
+  var rulesByReaction = groupRulesBy(rules, 'reactionName');
+
+  for (var reactionName in rulesByReaction) {
+    if (rulesByReaction.hasOwnProperty(reactionName)) {
+      checkRulesForReaction(rulesByReaction, reactionName, errors);
+    }
+  }
+}
+
+function checkRulesForReaction(rulesByReaction, reactionName, errors) {
+  var rules = rulesByReaction[reactionName];
+  checkChannelNamesAreSorted(rules, reactionName, errors);
+  checkForDuplicateRepositories(rules, reactionName, errors);
+  checkForDuplicateChannels(rules, reactionName, errors);
+  checkForDuplicateAllChannelRules(rules, reactionName, errors);
+}
+
+function checkChannelNamesAreSorted(rules, reactionName, errors) {
+  rules.forEach(function(rule, index) {
+    var sorted = sortIfUnsorted(rule.channelNames);
+
+    if (sorted) {
```

Comment:
Right, `sortIfUnsorted()` only returns a value if it had to sort the items. If that seems wonky or confusing, I can try to think of another interface (even though this is a strictly module-internal function).